contracts/contracts/buyback/Buyback.sol

```diff
@@ -51,25 +67,70 @@ contract Buyback is Governable {
      * @dev Execute a swap of OGN for OUSD via Uniswap or Uniswap compatible
      * protocol (e.g. Sushiswap)
      **/
-    function swap() external onlyVault {
-        if (uniswapAddr == address(0)) return;
-
+    function swap() external onlyVault nonReentrant {
         uint256 sourceAmount = ousd.balanceOf(address(this));
         if (sourceAmount < 1000 * 1e18) return;
+        if (uniswapAddr == address(0)) return;
+        // 97% should be the limits of our oracle errors.
+        // If this swap somtimes skips when it should succeed, that’s okay,
+        // the amounts will get get sold the next time this runs,
+        // when presumably the oracles are more accurate.
```

Comment:
This comment is very poetic, it feels like a haiku

Reply:
Hah, so it is!